app/controllers/expenses_controller.rb

```diff
@@ -22,6 +22,22 @@ def create
     end
   end
 
+  def edit
+    @categories = Category.subcategories
+    @expense = Expense.find(params[:id])
+  end
+
+  def update
+    @expense = Expense.find(params[:id])
+    expense_updater = UpdateExpenseAndReceipt.new(expense_params)
```

Comment:
I would prefer to see this class being instantiated with an existing `@expense`.
And then the `update` method would be called with newer params.
This would be a use case for the decorator pattern.